engine.go

```diff
@@ -156,6 +158,26 @@ type EvalResult struct {
 	Version string
 }
 
+// ToString decodes the value as a string.
```

Comment:
Seems like there's a strong overlap between this code (all of these `To...()` methods) and the private `exprTo....()` functions in `rule/expr.go`.  I wonder if we should just support a `To<go type>()` method on `rule.Value` types instead of doing these conversions in multiple places?